### PR TITLE
[MAPLv3] Remove functions from  submodule files

### PR DESCRIPTION
**Pull request by JulesKouatchou**
## Types of change(s)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)
- [ ] Refactor (no functional changes, no api changes)

## Checklist
- [ ] Tested this change with a run of GEOSgcm
- [x] Ran the Unit Tests (`make tests`)

## Description

Move "simple" `get_` and `set_` functions from submodule files into "parent" module files.

### CODE REVIEW

**Comment by mathomp4** [comment]
That CMakeLists conflict is above my knowledge. Perhaps @tclune had changes that came in while you were doing this change. 

@JulesKouatchou You might need to try and figure it out...

**Comment by JulesKouatchou** [comment]
I only removed .F90 file names in `generic3g/CMakeLists.txt`. I will try to check why there is a conflict.

**Comment by JulesKouatchou** [comment]
@mathomp4 and @tclune : the new file `generic3g/CMakeLists.txt` is fine.

File: generic3g/GriddedComponentDriver.F90 [hunk 1]
```diff
@@ -149,4 +149,48 @@ end subroutine add_import_coupler
 
    end interface
 
+   CONTAINS
+
+   module function get_clock(this) result(clock)
```
Comment (tclune) [inline generic3g/GriddedComponentDriver.F90:154]:
```suggestion
   function get_clock(this) result(clock)
```
`module` is not needed now that they live in the main module again.

**Review by tclune — CHANGES_REQUESTED** [review]
You need to remove the interface declarations for the procedures that are moved back into the main module.   And remove `module` in front of each actual procedure.

**Pull request closed without merging**